src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets

```diff
@@ -72,7 +72,8 @@ The .NET Foundation licenses this file to you under the MIT license.
     <ItemGroup Condition="'$(StaticICULinking)' == 'true'">
       <NativeLibrary Include="$(IntermediateOutputPath)/libs/System.Globalization.Native/build/libSystem.Globalization.Native.a"/>
       <DirectPInvoke Include="libSystem.Globalization.Native" />
-      <StaticICULibs Include="-Wl,-Bstatic -licuio -licutu -licui18n -licuuc -licudata -Wl,-Bdynamic" />
+      <StaticICULibs Include="-Wl,-Bstatic -licuio -licutu -licui18n -licuuc -licudata -lstdc++" />
+      <StaticICULibs Include="-Wl,-Bdynamic" Condition="'$(NativeLib)' != 'Static' and !@(LinkerArg->Contains('-static'))" />
```

Comment:
What do we need the linker args for when creating static library? Should the `'$(NativeLib)' != 'Static'` condition be on the whole ItemGroup?

Reply:
> What do we need the linker args for when creating static library?

Currently, we don't have any idiomatic property like `NativeLib=true` for executable. `NaitveLib=Static` expects `OutputType=Library` and throw error for `OutputType=Exe`.

> Should the '$(NativeLib)' != 'Static' condition be on the whole ItemGroup?

Yes, `StaticICULinking` is not needed with `NativeLib=Static` (libSystem.Globalization is already statically linked in that case).

Should we introduce `StaticExecutable` (boolean) property and disregard `<LinkerArg Include="-static"` in user project? We already do not support it properly. Then I will add `'$(NativeLib)' != 'Static' and '$(StaticExecutable)' != 'true'` to the ItemGroup.

Reply:
> What do we need the linker args for when creating static library?

Currently, we don't have any idiomatic property like `NativeLib=true` for executable. `NaitveLib=Static` expects `OutputType=Library` and throws error for `OutputType=Exe`.

> Should the '$(NativeLib)' != 'Static' condition be on the whole ItemGroup?

Yes, `StaticICULinking` is not needed with `NativeLib=Static` (libSystem.Globalization is already statically linked in that case).

Should we introduce `StaticExecutable` (boolean) property and disregard `<LinkerArg Include="-static"` in user project? We already do not support it properly. Then I will add `'$(NativeLib)' != 'Static' and '$(StaticExecutable)' != 'true'` to the ItemGroup.

Reply:
I've pushed an update.